File: PartyApp/app/src/main/java/com/example/alex/partyapp/servers/ServersAdapter.java
package com.example.alex.partyapp.servers;

import android.databinding.DataBindingUtil;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.ViewGroup;

import com.example.alex.partyapp.R;
import com.example.alex.partyapp.data.Server;

import java.util.ArrayList;
import java.util.List;

public class ServersAdapter extends RecyclerView.Adapter<ServerViewHolder> {
    private List<Server> data = new ArrayList<>();

    public void setData(List<Server> data) {
        this.data = data;
    }

    @Override
    public ServerViewHolder onCreateViewHolder(ViewGroup parent,
                                             int viewType) {
        return new ServerViewHolder(DataBindingUtil.inflate(LayoutInflater.from(parent.getContext()),
                R.layout.server_item, parent, false));
    }

    @Override
    public void onBindViewHolder(ServerViewHolder holder, int position) {
        holder.bind(data.get(position));
    }

    @Override
    public int getItemCount() {
        return data.size();
    }
}

File: app/src/main/java/com/example/androidparty/adapters/ServersAdapter.kt
package com.example.androidparty.adapters

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.databinding.DataBindingUtil
import androidx.recyclerview.widget.RecyclerView
import com.example.androidparty.R
import com.example.androidparty.databinding.ServerListItemBinding
import com.example.androidparty.viewmodel.ServerViewModel

class ServersAdapter(private var list: List<ServerViewModel>) : RecyclerView.Adapter<ServersAdapter.ServerViewHolder>() {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ServerViewHolder {
        val inflater = LayoutInflater.from(parent.context)
        val binding: ServerListItemBinding = DataBindingUtil.inflate(inflater, R.layout.server_list_item, parent, false)
        return ServerViewHolder(binding)
    }

    fun setData(data: List<ServerViewModel>){
        list = data
    }

    override fun getItemCount(): Int = list.size

    override fun onBindViewHolder(holder: ServerViewHolder, position: Int) {
        val item: ServerViewModel = list[position]
        holder.bind(item)
    }

    class ServerViewHolder(private val binding: ServerListItemBinding) : RecyclerView.ViewHolder(binding.root) {
        fun bind(item: ServerViewModel) {
            binding.server = item
            binding.executePendingBindings()
        }
    }
}